packages/manager/src/features/ObjectStorage/BucketLanding/BucketDetailsDrawer.test.tsx

```diff
@@ -200,3 +201,100 @@ describe('BucketDetailsDrawer: Legacy UI', () => {
     });
   });
 });
+
+describe('BucketDetailDrawer: Gen2 UI', () => {
+  const e3Bucket = objectStorageBucketFactoryGen2.build();
+  const e1Bucket = objectStorageBucketFactoryGen2.build({
+    endpoint_type: 'E1',
+  });
+
+  const region = regionFactory.build({
+    id: e3Bucket.region,
+  });
+
+  it('renders correctly when open', () => {
+    renderWithThemeAndHookFormContext({
+      component: (
+        <BucketDetailsDrawer
+          onClose={mockOnClose}
+          open={true}
+          selectedBucket={e3Bucket}
+        />
+      ),
+      options: {
+        flags: { objMultiCluster: false, objectStorageGen2: { enabled: true } },
+      },
+    });
+
+    expect(screen.getByText(e3Bucket.label)).toBeInTheDocument();
+    expect(screen.getByTestId('createdTime')).toHaveTextContent(
+      'Created: 2019-12-12'
+    );
+    expect(screen.getByTestId('endpointType')).toHaveTextContent(
+      `Endpoint Type: E3`
+    );
+    expect(screen.getByTestId('cluster')).toHaveTextContent(region.id);
+    expect(screen.getByText(e3Bucket.hostname)).toBeInTheDocument();
+    expect(screen.getByText('1 MB')).toBeInTheDocument();
+    expect(screen.getByText('103 objects')).toBeInTheDocument();
+  });
+
+  it("doesn't show the CORS switch for E2 and E3 buckets", async () => {
+    const { getByText } = renderWithThemeAndHookFormContext({
+      component: (
+        <BucketDetailsDrawer
+          onClose={mockOnClose}
+          open={true}
+          selectedBucket={e3Bucket}
+        />
+      ),
+      options: {
+        flags: { objMultiCluster: false, objectStorageGen2: { enabled: true } },
+      },
+    });
+
+    expect(
+      getByText(
+        /CORS \(Cross Origin Sharing\) is not available for endpoint types E2 and E3./
+      )
+    ).toBeInTheDocument();
+  });
+
+  it('renders the Bucket Rate Limit Table for E2 and E3 buckets', async () => {
+    const { getByTestId } = renderWithThemeAndHookFormContext({
+      component: (
+        <BucketDetailsDrawer
+          onClose={mockOnClose}
+          open={true}
+          selectedBucket={e3Bucket}
+        />
+      ),
+      options: {
+        flags: { objMultiCluster: false, objectStorageGen2: { enabled: true } },
+      },
+    });
+
+    const rateLimitTable = getByTestId('bucket-rate-limit-table');
```

Comment:
This test is failing since `isObjectStorageGen2Enabled` is derived from an RQ hook and is initially `undefined`. Awaiting via `await findByTestId` will wait for the query to resolve.

```suggestion
    const { findByTestId } = renderWithTheme(
      <BucketDetailsDrawer
        onClose={mockOnClose}
        open={true}
        selectedBucket={e3Bucket}
      />,
      {
        flags: { objMultiCluster: false, objectStorageGen2: { enabled: true } },
      }
    );

    const rateLimitTable = await findByTestId('bucket-rate-limit-table');
```

Reply:
Thank you so much for pointing this out! I've made the changes in 8974ea41a7d26f46fdb12d9f33c0e56cec2513c0